src/server/scripts/Spells/spell_hunter.cpp

```diff
@@ -1011,10 +1012,47 @@ class spell_hun_tnt : public SpellScriptLoader
             return new spell_hun_tnt_AuraScript();
         }
 };
+// -186257 - Aspect of the cheetah
+class spell_hun_aspect_cheetah : public SpellScriptLoader
+{
+    public:
+        spell_hun_aspect_cheetah() : SpellScriptLoader("spell_hun_aspect_cheetah") { }
+
+        class spell_hun_aspect_cheetah_AuraScript : public AuraScript
+        {
+
+            PrepareAuraScript(spell_hun_aspect_cheetah_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo
+                ({
+                    SPELL_HUNTER_ASPECT_CHEETAH_SLOW
+                });
+            }
+
+            void HandleOnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                GetCaster()->CastSpell(GetCaster(), SPELL_HUNTER_ASPECT_CHEETAH_SLOW, true);
```

Comment:
GetCaster() can be null in AuraScripts.
If Target of Aura == Caster you can use GetTarget() instead, because in AuraScripts GetTarget() cant be null.